CHANGELOG.md

```diff
@@ -20,10 +20,13 @@ and this project adheres to
 - Upgrade rust compiler to 1.83.0
 - Upgrade Go compiler to 1.24.0
 - Add charconv lib build in boost
+- Update Go compiler to 1.24.1
+- Use avahi as mdns provider
```

Comment:
I'd like to do more with mDNS -- what were we using before? What are the tradeoffs between avahi and systemd-resolved?

Reply:
avahi supports more e.g. traditional zeroconf situations as well, systemd-resolved handles global DNS resolution and LLMNR well but I am not sure how good it is with mDNS. they can coexist with systemd-resolved handling general DNS resolution while Avahi provides more comprehensive local service discovery and advertising.